File: apps/api/src/routes/_p3Register.ts
/**
 * P3 router + worker registration helper.
 *
 * Isolates all Phase 3 (P3 Lite — Matrimony AI Assistant + Vendor Engine
 * foundation) wiring behind two functions so apps/api/src/index.ts only
 * changes by three lines (import + mount call + worker call). Reduces
 * merge-conflict surface with the parallel P2 session that also edits
 * index.ts.
 */
import type { Express } from 'express';
import type { Worker } from 'bullmq';
import { assistantRouter } from './assistant.js';
import { vendorEngineRouter } from './vendorEngine.js';
import { referralRouter } from './referral.js';
import {
  registerVendorAvailabilityRefreshWorker,
  scheduleVendorAvailabilityRefreshJob,
} from '../jobs/vendorAvailabilityRefreshJob.js';

export function registerP3Routes(app: Express): void {
  app.use('/api/v1/assistant', assistantRouter);
  app.use('/api/v1/vendor-engine', vendorEngineRouter);
  app.use('/api/v1/referral', referralRouter);
}

export function registerP3Workers(workers: Array<{ close(): Promise<void> }>): void {
  const worker: Worker = registerVendorAvailabilityRefreshWorker();
  workers.push(worker);
  void scheduleVendorAvailabilityRefreshJob();
}

File: apps/ai-service/src/routers/_p3_register.py
"""
P3 router registration helper.

Isolates all P3-introduced FastAPI router mounts behind a single function so
src/main.py only changes by two lines (import + call). Future P3 routers are
added here, not in main.py — keeps the shared file low-churn.
"""

from __future__ import annotations

from fastapi import FastAPI

from src.routers.assistant import router as assistant_router
from src.routers.embedding import router as embedding_router


def register_p3_routers(app: FastAPI) -> None:
    """Mount all Phase 3 P3 routers on the given app."""
    app.include_router(assistant_router)
    app.include_router(embedding_router)
